src/js/helpers/oboeWrapper.js

```diff
@@ -1,28 +1,53 @@
 var oboe = require("oboe");
 
 var oboeWrapper = function (req) {
+  var response = {
+    status: null,
+    body: null
+  };
+
+  var parseResponse = function (xhr) {
+    response.status = xhr.statusCode;
+    try {
+      response.body = JSON.parse(xhr.body);
+    } catch (e) {
+      response.body = xhr.body;
+    }
+  };
+
   var api = oboe(req)
     .start(function (status) {
       this.status = status;
     });
 
   api.success = function (callback) {
-    this.done(function (payload) {
+    this.done(function (json) {
       if (!this.status.toString().match(/^2[0-9][0-9]$/)) {
         return;
       } else {
-        callback(payload);
+        response.status = this.status;
+        response.body = json;
+        callback(response);
       }
     });
     return this;
   };
 
   api.error = function (callback) {
+    // Non 200 statuses
+    this.done(function (error) {
+      if (this.status.toString().match(/^[^2]/)) {
```

Comment:
Seems pretty like overkill to me. :)
Why that [0-9][0-9] ? It just covers _all_ cases, so it's obsolete.
And if the status code isn't an Number there is something other totally wrong, hehe.